projects/python-client/src/datapane/client/config.py

```diff
@@ -160,13 +176,11 @@ def upgrade_config_format(self):
 
                     capture("CLI Login", config=self, with_token=True)
 
-            self.version = 5
+            self.version = LATEST_VERSION
             self.save()
-        elif self.version == 4:
-            self.version = 5
+        elif self.version in (4, 5):
+            self.version = LATEST_VERSION
             self.save()  # trigger removal of _env & _pass from file
-        elif self.version == 5:
-            pass  # current
```

Comment:
If we can assumed everyone's on v5 then we don't need this. Which version was v4, and do we have stats for how many active users are still on it? I guess we could add a version check to `ensure_config_file` and temporarily publish some upgrade instructions somewhere to tell people to downgrade to 0.15.4, run `datapane ping` then upgrade again?

Reply:
Yeah I think let's do that - we can assume everyone is on v5 I believe (it was released about 4 months ago). I think a fallback error message for < v5 sounds like a great plan - let's do that!

Reply:
Following this idea through, we need to convert the file to an ini so we can load it using configparser before we can find out what version it is. By then it will be too late to roll back to v5 without needing to undo the yaml->ini conversion. Not impossible of course, but does complicate the upgrade procedure.

Are we ok to just say if version < 5, raise an error saying something like "Config file version too old to load, please log in again" and just reset the config?

Reply:
Yes, exactly - that would be the path, i.e. if < 5, then delete the old config and create a new / fresh one (we shouldn't explicitly require re-login as some folks make be upgrading from an older version and never using the cloud features).

We bumped to v5 in 0.15.0, released at end Aug, which would be a major version that required users to upgrade to continue using DPCloud. We can assume that most DPCloud users will have updated in the last 3 months to continue using the Cloud features (or they were using locally in which case it isn't a problem).

Reply:
Resolving - to pick up removing support of < v5 as part of 0.16 release